### ISSUE TITLE
[FEA] Using cupy with  __array_function__  for functions not currently supported 

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#2718]
### Using cupy with  __array_function__  for functions not currently supported 

In the initial issue, https://github.com/rapidsai/cudf/issues/1728 there was discussion (https://github.com/rapidsai/cudf/issues/1728#issuecomment-497351710) around using cupy for functions which we don't have support currently for numerical data (non-null  (int/bool) and float )

This was not initially implemented because we had some  packaging issues surrounding CuPy which i think have gone away so we can tackle this. 

This will unlock a lot of features for us, see (https://docs-cupy.chainer.org/en/stable/reference/comparison.html ) without much code changes and help a lot with interoperability.

**Comment by VibhuJawa** [comment]
P.S: I would love to tackle this as i wrote the initial array_function implementation

**Comment by kkraus14** [comment]
@VibhuJawa do you know where we landed on this?

**Comment by VibhuJawa** [comment]
> @VibhuJawa do you know where we landed on this?

IIRC, we were waiting on the packaging issues but they are no longer  present so we can probably tackle this on now.

I am unsure about prioritization as the work around is just using a cupy function on the series directly but this may still be a nice to have  . 

For additional context [pr](https://github.com/rapidsai/cudf/pull/2148 ) took a hacky stab at this .
